chucks-pick3-web/angular-app/src/app/components/log-in.component/log-in.component.ts

```diff
@@ -7,12 +10,34 @@ import {Component, OnInit} from '@angular/core';
 })
 
 export class LogInComponent implements OnInit {
-  constructor() {
+
+
+  public errorMessage: string;
+
+  constructor(public formValidationService: FormValidationService,
+              public memberService: MemberService,
+              public router: Router) {
   }
 
   ngOnInit() {
     const howToActive = document.getElementById('howTo');
     howToActive.classList.remove('active');
     howToActive.classList.add('allow-hover');
   }
+
+  public onSubmit() {
+    const isFormValid = this.formValidationService.validateForm();
+    const isAccountCreated = this.memberService.logInSuccessful();
```

Comment:
I see what you are trying to do here.  Actually `isAccountCreated` property is not an accurate description of the `logInSuccessful() method`.  Consider renaming it to something more intuitive.  Rename `this.memberSErvice.logInSuccessful() method` to something that accurately reflects what the user is trying to do once the submit the login form.  Take a moment and think about it before you come up with the names.  Lastly the `logInSuccessful() method` will take some time to execute as it has to make a round trip to the AWS cloud and back; so we only want to call it, if and only if, the form is valid.  Consider calling `logInSuccessful() method` only if the form is valid.

Reply:
Changed the location of the logInSuccessful within the nested if statements. Also renamed both the method and property to:
 const isAccountAccessed = this.memberService.accessLogIn();